openhtf/__init__.py

```diff
@@ -161,14 +162,27 @@ def plug_type_map(self):
     return plug_type_map
 
 
-def SetupFramework():
-  """Sets up various bits of the framework. Only needs to be called once."""
-  try:
-    sys.argv = FLAGS(sys.argv)  # parse flags
-  except gflags.FlagsError, e:  # pylint: disable=invalid-name
-    print '%s\nUsage: %s ARGS\n%s' % (e, sys.argv[0], FLAGS)
-    sys.exit(1)
+def CreateArgParser():
+  """Creates an argparse.ArgumentParser for parsing command line flags."""
+  parser = argparse.ArgumentParser('OpenHTF-based testing')
+  # openhtf.io.user_input
+  # Option 1:
+  parser.add_argument('--prompt_timeout_s', type=int,
+                      help='User prompt timeout in seconds.')
+  # Option 2:
+  parser = argparse.ArgumentParser('OpenHTF-based testing', parents=[
+      user_input.DEFAULT_TIMEOUT_S_PARSER])
+
+
+  return parser
 
+
+@functions.RunOnce
+def SetupFramework(arguments=None):
+  """Sets up various bits of the framework. Only needs to be called once."""
+  parser = arguments or CreateArgParser().parse_args()
+  # Option 3:
```

Comment:
I'm leaning towards option 3.

Option 1 is the worst because it requires the framework entry logic and the `user_input` submodule to know about _each other_.

Option 2 is slightly better because the entry logic is the relationship only goes one way, but the API for doing it is less straightforward.

Option 3 is also a one-way relationship, and is slightly more straightforward, at least when you're looking at this end of it.

The fourth (and possibly best but also most effort) option would be to redesign prompt management so it doesn't need to use a singleton that's created at import time, which is where the complexity seems to stem from, and I'll take the blame for doing it this way to begin with. I was following a pattern established by Python's own `logging` module.